napari_aicsimageio/core.py

```diff
@@ -41,6 +43,86 @@ def _get_full_image_data(img: AICSImage, in_memory: bool) -> Optional[xr.DataArr
     return None
 
 
+# Function to handle multi-scene files.
+def _get_scenes(img: AICSImage, in_memory: bool) -> None:
+
+    # Create the list widget and populate with the ids & scenes in the file
+    list_widget = QListWidget()
+    for i, scene in enumerate(img.scenes):
+        list_widget.addItem(f"{i} :: {scene}")
+    viewer = napari.current_viewer()
+    viewer.window.add_dock_widget(list_widget, area="right", name="Scene Selector")
+
+    # Function to create image layer from a scene selected in the list widget
+    def open_scene(item) -> None:
+        scene_text = item.text()
+
+        # Use scene indexes to cover for duplicate names
+        scene_index = int(scene_text.split(" :: ")[0])
+        img.set_scene(scene_index)
+        if DimensionNames.MosaicTile in img.reader.dims.order:
+            try:
+                if in_memory:
+                    data = img.reader.mosaic_xarray_data
+                else:
+                    data = img.reader.mosaic_xarray_dask_data
+
+            # Catch reader does not support tile stitching
+            except NotImplementedError:
+                print(
+                    "AICSImageIO: Mosaic tile stitching "
+                    "not yet supported for this file format reader."
+                )
+        else:
+            if in_memory:
+                data = img.reader.xarray_data
+            else:
+                data = img.reader.xarray_dask_data
+        meta = _get_meta(data, img)
+        viewer.add_image(data, name=scene_text, metadata=meta, scale=meta["scale"])
+
+    list_widget.currentItemChanged.connect(open_scene)
+
+
+# Function to get Metadata to provide with data
+def _get_meta(data, img) -> Dict[str, Any]:
```

Comment:
This `data` and `img` needs to be typed. I believe they should be `data: xr.DataArray, img: AICSImage`